File: 加密保护/The Enigma Protector v2.70 Build 20110629/Examples/HardwareID/Bcb/test_unit.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "test_unit.h"

// Enigma API respondence ---------------------------------------------------
#include "../../../EnigmaSDK/Bcb/enigma_ide.h"
#pragma link "../../../EnigmaSDK/Bcb/enigma_ide.lib"
// --------------------------------------------------------------------------

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TfrmRegistration *frmRegistration;

  char* pcUserInfo = NULL;
  char* pcKey = NULL;

//---------------------------------------------------------------------------
__fastcall TfrmRegistration::TfrmRegistration(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TfrmRegistration::btnExitClick(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------
void __fastcall TfrmRegistration::FormShow(TObject *Sender)
{
  char	buffer[1024];

  eHardware->Text = EP_RegHardwareID();
  if (EP_RegLoadAndCheckKey())
  {
    if (EP_RegLoadKey(&pcUserInfo, &pcKey))
    {
      lbUserInfo->Caption = pcUserInfo;
      mKey->Text = pcKey;
    }
  } else
  {
    buffer[0] = 0;
    lstrcat(buffer, "*************   Reminder window   *************\n\n\n");
    lstrcat(buffer, "   Register software to remove this message!   \n");
    lstrcat(buffer, "    For registration visit www.YourProduct.com \n\n");
    ShowMessage(buffer);
  }
}
//---------------------------------------------------------------------------
void __fastcall TfrmRegistration::btnRegisterClick(TObject *Sender)
{
  char Key[255];
  char UserInfo[255];

  StrPCopy(UserInfo, eUserInfo->Text);
  StrPCopy(Key, eKey->Text);

  if (EP_RegCheckAndSaveKey((char*)UserInfo, (char*)Key))
  {
    lbUserInfo->Caption = eUserInfo->Text;
    mKey->Text = eKey->Text;
    ShowMessage("Thanks for registration");
  } else
  {
    ShowMessage("Invalid key");
  }
}
//---------------------------------------------------------------------------
void __fastcall TfrmRegistration::btnDeleteKeyClick(TObject *Sender)
{
  EP_RegDeleteKey();
}
//---------------------------------------------------------------------------

File: 加密保护/The Enigma Protector v2.70 Build 20110629/Examples/RegistrationUnicode/Delphi_2009UP/test_unit.pas
unit test_unit;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, ExtCtrls, StdCtrls;

type
  TfrmRegistration = class(TForm)
    btnRegister: TButton;
    btnExit: TButton;
    Label1: TLabel;
    eHardware: TEdit;
    eUserInfo: TEdit;
    eKey: TEdit;
    Label3: TLabel;
    Label4: TLabel;
    Label2: TLabel;
    Label5: TLabel;
    lbUserInfo: TLabel;
    Bevel1: TBevel;
    btnDeleteKey: TButton;
    mKey: TMemo;
    procedure btnExitClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
    procedure btnRegisterClick(Sender: TObject);
    procedure btnDeleteKeyClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmRegistration: TfrmRegistration;

implementation

uses enigma_ide;

{$R *.dfm}

procedure TfrmRegistration.btnExitClick(Sender: TObject);
begin
  Application.Terminate;
end;

procedure TfrmRegistration.FormShow(Sender: TObject);
var
  pcKey, pcUserInfo : PWideChar;
begin
  eHardware.Text := EP_RegHardwareIDW;
  if EP_RegLoadAndCheckKey then
  begin
    if EP_RegLoadKeyW(pcUserInfo, pcKey) then
    begin
      lbUserInfo.Caption := pcUserInfo;
      mKey.Text := pcKey;
    end;
  end else
    ShowMessage('*************   Reminder window   *************'#13#10#13#10#13#10 +
                '   Register software to remove this message!   '#13#10 +
                '    For registration visit www.YourProduct.com    '#13#10#13#10);
end;

procedure TfrmRegistration.btnRegisterClick(Sender: TObject);
var
  pcKey, pcUserInfo : PWideChar;
begin
  if EP_RegCheckAndSaveKeyW(PWideChar(eUserInfo.Text), PWideChar(eKey.Text)) then
  begin
    if EP_RegLoadKeyW(pcUserInfo, pcKey) then
    begin
      lbUserInfo.Caption := pcUserInfo;
      mKey.Text := pcKey;
      ShowMessage('Thanks for registration');
    end;
  end else ShowMessage('Invalid key');
end;

procedure TfrmRegistration.btnDeleteKeyClick(Sender: TObject);
begin
  EP_RegDeleteKey;
end;

end.
